### PR TITLE
Implements todo-due-dates spec

### PR DESCRIPTION
**Pull request by scalvert**
Implements spec from #31

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
+}
```
Comment (rwjblue) [inline .vscode/settings.json:3]: Editor specific files should not be checked in 😛

File: package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "@types/tmp": "^0.2.0",
     "@typescript-eslint/eslint-plugin": "^4.2.0",
     "@typescript-eslint/parser": "^4.2.0",
+    "date-fns": "^2.16.1",
```
Comment (rwjblue) [inline package.json:28]: Seems like this is used "in production" (and should be a dep), no?

File: .vscode/settings.json (hunk 1 above)
Comment (scalvert) [reply .vscode/settings.json:3]: Oops.

File: package.json (hunk 2 above)
Comment (scalvert) [reply package.json:28]: It's just used in tests for the date comparison, so no. It's just a dev dep.

Comment (rwjblue) [reply package.json:28]: Ah. Gotcha, sorry thought I saw it outside tests too.

**Pull request merged**